test/snippet/argument_parser/custom_enumeration.cpp

```diff
@@ -0,0 +1,47 @@
+#include <seqan3/argument_parser/all.hpp>
+#include <seqan3/range/views/all.hpp>
+#include <seqan3/range/views/get.hpp>
+
+namespace your_namespace
+{
+
+enum class Foo
+{
+    one,
+    two,
+    three
+};
+
+// Define a global variable to reuse in the customisation and the seqan3::value_list_validator.
+std::unordered_map<std::string_view, Foo> naming{{{"one", Foo::one}, {"two", Foo::two}, {"three", Foo::three}}};
+
+// Specialise a mapping from an identifying string to the respective value of your type Foo.
+auto & enumeration_names(Foo)
+{
+    return naming;
+}
```

Comment:
I thought it's better because the map then wont be copied every time I call `enumeration_names`, I didn't think about linkage

Reply:
> I thought it's better because the map then wont be copied every time I call enumeration_names

But you are not supposed to call `enumeration_names` at all, the CPO is the interface. And the CPO is variable that is initialised only once.